include/serialport.h

```diff
@@ -47,116 +50,143 @@
 #include "hardware.h"
 #endif
 
-// Serial port interface 
+// Serial port interface
+
+#define SERIAL_MAX_FIFO_SIZE 256
+/* Note: Almost all DOS-era universal asynchronous receiver-transmitter
+ *       (UART)'s permitted up to a 16-byte receive and transmit
+ *       first-in/first-out (FIFO) buffer, however some specialty
+ *       controller cards allowed up to 64-bytes and later 256-bytes.
+ */
 
 class MyFifo {
 public:
-	MyFifo(Bitu maxsize_) {
-		maxsize=size=maxsize_;
-		pos=used=0;
-		data=new Bit8u[size];
-	}
-	~MyFifo() {
-		delete[] data;
-	}
-	INLINE Bitu getFree(void) {
-		return size-used;
-	}
-	bool isEmpty() {
-		return used==0;
-	}
-	bool isFull() {
-		return (size-used)==0;
-	}
+	MyFifo(const MyFifo &) = delete;            // prevent copying
+	MyFifo &operator=(const MyFifo &) = delete; // prevent assignment
 
-	INLINE Bitu getUsage(void) {
-		return used;
+	MyFifo(size_t n) : data(n, 0), maxsize(n), size(n)
+	{
+		assert(n <= SERIAL_MAX_FIFO_SIZE);
```

Comment:
- rename array size `maxsize_` with `n`
- `data` is now a vector which we size with `n`
- `pos` and `used` are now initialized to `0` so don't need to be part of the constructor.
- `maxsize` and `size` members are assigned like in original, and all logic in this class it kept the same.